exercise/staff_views.py

```diff
@@ -271,6 +292,28 @@ def post(self, request, *args, **kwargs):
         return self.redirect(self.submission.get_inspect_url())
 
 
+class RegradeView(ExerciseBaseView, BaseRedirectView):
+    # For the time being only superusers have access.
+    # Later this will be loosened to ACCESS.TEACHER or ACCESS.ASSISTANT.
+    access_mode = ACCESS.SUPERUSER
+    template_name = "exercise/staff/regrade_submissions.html"
+
+    def get_common_objects(self) -> None:
+        self.regrade_type = self.request.GET.get('regrade', None)
+
+        if self.exercise.regrade_task:
+            self.result = AsyncResult(self.exercise.regrade_task)
+            self.note("result")
+        else:
+            job = regrade_exercises.delay(
```

Comment:
Are you creating a new background task in an HTTP GET request? Very suspicious. In principle, the database is never modified in GET requests except for some kind of logging. There are good reasons for that. GET is a so-called "safe" method and people assume it is safe to use (idempotent). 

At any rate, `get_common_objects` is not the right method for creating new Celery tasks.

-----------------------------------------------

I remember an old blog post that describes how Google Search bot crawler regularly deleted one site's whole database because items were deleted with GET requests that have query param `action=delete`. Google just followed links with GET requests like they are supposed to do.